test/parallel/test-timers-api-refs.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+// don't verify the globals for this test
+common.globalCheck = false;
+
+// try overriding global APIs to make sure
+// they're not relied on by the timers
+global.clearTimeout = assert.fail;
```

Comment:
`clearTimeout(..)`, `unenroll(..)` and `active(..)` are the only methods that are called from the code itself, which means they are the only ones that could break code if allowed to be overwritten without a safe internal reference. That's why I went the conservative route.

We could apply the fix to all API methods, but if they aren't being called, it seems like unnecessary extra code.